### PR TITLE
Feature/APMI3255 iOS: options factory update

### PR DESCRIPTION
**Pull request by Piyush23786**
APMI-3255:-
1. Set Location method that sets global attributes (only 2 numbers - latitude and longitude)
2. Bundle identifier (CF Bundle) that is detected and added automatically.
3. Feature flag for slow/frozen renders
4. Contents attribute added.

### CODE REVIEW

File: SplunkRumWorkspace/SplunkRum/SplunkRum/Attribute.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+       static let SPAN_NAME_APP_ENTERFOREGROUND      = "EnterForeground"
+       static let SPAN_NAME_SCREEN_NAME_CHANGE       = "screen name change"
+       static let SPAN_NAME_APP_START                = "AppStart"
+       static let SPAN_NAME_SESSIONID_CHANGE         = "sessionId.change"
+       static let SPAN_NAME_SHOWVC                   = "ShowVC"
+       static let SPAN_NAME_ACTION                   = "action"
+       static let SPAN_NAME_SPLUNKRUM_INITIALIZE     = "SplunkRum.initialize"
+       static let SPAN_NAME_MANUAL_SPAN              = "manualSpan"
+       static let SPAN_NAME_SLOWRENDERS              = "slowRenders"
+       static let SPAN_NAME_FROZENRENDERS            = "frozenRenders"
```
Comment (seemk) [inline SplunkRumWorkspace/SplunkRum/SplunkRum/Attribute.swift:56]: I think these different sections should be separated. Currently there are attribute key names, component attribute value names and span name values grouped together under `Attribute`.

File: SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift [hunk 2]
```diff
@@ -423,4 +275,19 @@ var splunkRumInitializeCalledTime = Date()
...
+            setGlobalAttributes([Attribute.LOCATION_LONGITUDE_KEY: longitude])
+            setGlobalAttributes([Attribute.LOCATION_LATITUDE_KEY: latitude])
+    }
+
+    /**
+      Set up the deployment environment for RUM instance. This will be passed along as a span
+     */
+    @objc public class func deploymentEnvironment(environment: String) {
+            setGlobalAttributes([Attribute.DEPLOYMENT_ENVIRONMENT: environment])
+    }
```
Comment (seemk) [inline SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift:292]: Deployment environment is only set at startup, so this method can be removed

File: SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift [hunk 3]
```diff
@@ -423,4 +275,19 @@ var splunkRumInitializeCalledTime = Date()
...
         return initialized
     }
+
+    /**
+       Updates the current location. The latitude and longitude will be appended to every span and event.
+     */
+    @objc public class func setLocation(latitude: Double, longitude: Double) {
+            setGlobalAttributes([Attribute.LOCATION_LONGITUDE_KEY: longitude])
+            setGlobalAttributes([Attribute.LOCATION_LATITUDE_KEY: latitude])
+    }
```
Comment (seemk) [inline SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift:285]: 👍 But can do it in a single call to `setGlobalAttributes`

File: SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift [hunk 4]
```diff
@@ -24,155 +23,6 @@ import WebKit
...
 
-/**
- Default maximum size of the disk cache in bytes.
- */
-public let DEFAULT_DISK_CACHE_MAX_SIZE_BYTES: Int64 = 25 * 1024 * 1024
-
-/**
- Optional configuration for SplunkRum.initialize()
- */
-@objc public class SplunkRumOptions: NSObject {
```
Comment (seemk) [inline SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift:35]: Is the removal from this file transparent to end users? I don't know enough here, but say someone is importing `SplunkRumOptions`, is there a chance the moving of this class will break someone's code?

File: SplunkRumWorkspace/SplunkRum/SplunkRum/Attribute.swift (hunk 1 above)
Comment (Piyush23786) [reply SplunkRumWorkspace/SplunkRum/SplunkRum/Attribute.swift]: separated as a KeyValue,Component & SpanName.

File: SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift (hunk 2 above)
Comment (Piyush23786) [reply SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift]: Removed

File: SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift (hunk 3 above)
Comment (Piyush23786) [reply SplunkRumWorkspace/SplunkRum/SplunkRum/SplunkRum.swift]: Done in a single call.

**Pull request closed without merging**